src/main/java/net/snowflake/client/jdbc/SnowflakeBasicDataSource.java

```diff
@@ -231,4 +231,12 @@ public void setPrivateKeyFile(String location, String password) {
   public void setTracing(String tracing) {
     this.properties.put("tracing", tracing);
   }
+
+  public Properties getProperties() {
+    return this.properties;
+  }
+
+  public void setProperties(Properties props) {
```

Comment:
I'm not sure if we should expose the getters and setters for properties - the better would be to add missing setSth methods or create single method `setProperty(String name, String value)`
The frist option would be better since we would be able to add additional checks on provided values

Reply:
I just thought it's easier than constantly maintaining setters and getters for every single [parameter](https://docs.snowflake.com/en/developer-guide/jdbc/jdbc-parameters) that's introduced in the future.
If you still don't like that approach then I'll go back and add setters for whatever missing parameters there are.

Reply:
I think we should have good setters with validation of possible values for each parameter and getters are not necessary for the datasource - having the Properties field inside is only the implementation detail that we should not share because we won't be able to easy maintain internal state - e.g. someone could set in properties `authenticator` to `OAUTH`, but this class maintain separate field `authenticator`

Reply:
Done.